### PR TITLE
fix flaky test TestMarkMessagesSeenMarksNotificationsRead

### LINKED ISSUE #4404: flaky test - TestMessengerActivityCenterMessageSuite/TestMarkMessagesSeenMarksNotificationsRead
```
2023-11-30T16:43:07.378Z	DEBUG	protocol/messenger.go:1873	running shutdown task	{"n": 8}
    messenger_activity_center_test.go:398: 
        	Error Trace:	/home/jenkins/workspace/status-go_prs_tests_PR-4325/protocol/messenger_activity_center_test.go:398
        	Error:      	"[%!s(*protocol.ActivityCenterNotification=&{[255 20 151 199 159 194 254 194 207 52 225 15 17 210 132 102 64 124 138 236 134 181 167 110 31 2 44 13 209 160 177 17] 0x02cd21555e97d1d98ced52b1f36a5e6d4f8a2b8233b2b603212a4ad00fbe74b9df93cc0ef0-a9b8-4126-9e4f-f7a555e753c8 0x02cd21555e97d1d98ced52b1f36a5e6d4f8a2b8233b2b603212a4ad00fbe74b9df 0 status-core 0x040e64269f73b10ef0182611cf65082860e7036228d400b4faf046b14dae4bdc1442428f61dd6f0a936a78423ef229800aa97b9cf8e6f5e43003d5bac7a0ccd115 3 0xc020d40480 0xc020d406c0 <nil> 1701362587000 true false false false 0 1701362587379 []})]" should have 2 item(s), but has 1
```

```
messenger_activity_center_test.go:423: 
        	Error Trace:	/home/jenkins/workspace/status-go_prs_tests_PR-4508/protocol/messenger_activity_center_test.go:423
        	Error:      	"[%!s(*protocol.ActivityCenterNotification=&{[5 100 12 207 123 185 42 41 224 10 18 255 114 74 34 231 99 245 48 221 25 104 143 220 6 141 147 255 239 218 18 41] 0x031cb5e108c37371fd4234adcc6880f5e0b6c35349c891b9bf01fd63a66d276f5a8a5dafb5-013d-4284-9d07-3dabe585d02c 0x031cb5e108c37371fd4234adcc6880f5e0b6c35349c891b9bf01fd63a66d276f5a 0 status-core 0x048b3030d6a3acecdb7a7197341dff361364a286a670a9b773cb2819b5a88a95eb54fe4287425cf7f9990ecf5bbd85556abcc8641da1293e06b39a1e2095f12f02 3 0xc017714b40 0xc017714d80 <nil> 1704293806000 true false false false 0 1704293806996 []})]" should have 2 item(s), but has 1
        	Test:       	TestMessengerActivityCenterMessageSuite/TestMarkMessagesSeenMarksNotificationsRead
```

### PR DESCRIPTION
**Pull request by qfrank**
Closes #4404 

status: ready.

### CODE REVIEW

File: protocol/messenger_activity_center_test.go [hunk 1]
```diff
@@ -376,16 +376,24 @@ func (s *MessengerActivityCenterMessageSuite) prepareCommunityChannelWithMention
...
+	messages := response.Messages()
+	// One message is the community message, the other is the reply to the community message
+	s.Require().Len(messages, 2)
+
 	// There is an extra message with reply
-	if response.Messages()[0].ID == response.ActivityCenterNotifications()[0].Message.ID {
-		replyMessage = response.Messages()[0]
-	} else if response.Messages()[1].ID == response.ActivityCenterNotifications()[0].Message.ID {
-		replyMessage = response.Messages()[1]
+	if messages[0].GetResponseTo() != "" {
```
Comment (igor-sirotin) [inline protocol/messenger_activity_center_test.go]: Maybe we could just use the `response.messages` map directly?

```go
replyMessage, ok := response.messages[replyNotification.Message.ID]
s.Require().True(ok)
s.Require().NotNil(replyMessage)
```

**Review by igor-sirotin — COMMENTED** [review]
Nice catch!
I believe this is not the only place with such problem. We should probably check all usages of `Messages()`.

Also, please checkout my inline comment. I think it's an easier way when the MessageID is known.

File: protocol/messenger_activity_center_test.go (hunk 1 above)
Comment (qfrank) [reply protocol/messenger_activity_center_test.go]: applied

Comment (qfrank) [reply protocol/messenger_activity_center_test.go]: > We should probably check all usages of Messages()

found a lot of usages, i'll take a look at this tomorrow, thank you @igor-sirotin

**Review by igor-sirotin — APPROVED** [review]

File: protocol/messenger_activity_center_test.go [hunk 2]
```diff
@@ -376,16 +375,16 @@ func (s *MessengerActivityCenterMessageSuite) prepareCommunityChannelWithMention
...
-		replyMessage = response.Messages()[0]
-	} else if response.Messages()[1].ID == response.ActivityCenterNotifications()[0].Message.ID {
-		replyMessage = response.Messages()[1]
-	} else {
-		s.Error(errors.New("can't find corresponding message in the response"))
-	}
+	// One message is the community message, the other is the reply to the community message
+	s.Require().Len(response.Messages(), 2)
+
+	replyMessage, ok := response.messages[replyNotification.Message.ID]
```
Comment (MishkaRogachev) [inline protocol/messenger_activity_center_test.go:384]: Nice!

**Review by MishkaRogachev — APPROVED** [review]

**Pull request merged**